src/express/routes/articlesRouter.js

```diff
@@ -3,13 +3,57 @@
 const {Router} = require(`express`);
 const {URL} = require(`./../../service/constants`);
 const articlesRouter = new Router();
+const {getAPI} = require(`./../api`);
+const upload = require(`./../storage`);
+
+const api = getAPI();
+
+const collectCategory = async (body) => {
+  const categories = await api.getCategories();
+  const selectedCategories = [];
+  categories.forEach((elem, index) => {
+    if (body[`category-${index}`]) {
+      selectedCategories.push(elem);
+    }
+  });
+  return selectedCategories;
+};
 
 articlesRouter.get(`${URL.ARTICLESURL.CATEGORY}/:id`, (request, response) =>
   response.render(`articles-by-category`));
-articlesRouter.get(URL.ARTICLESURL.ADD, (request, response) => response.render(`new-post`));
-articlesRouter.get(`${URL.ARTICLESURL.EDIT}/:id`, (request, response) =>
-  response.render(`post`));
+articlesRouter.get(URL.ARTICLESURL.ADD, async (request, response) => {
+  const categories = await api.getCategories();
+  response.render(`new-post`, {categories});
+});
+articlesRouter.get(`${URL.ARTICLESURL.EDIT}/:id`, async (request, response) => {
+  const article = await api.getArticle(request.params.id);
+  const categories = await api.getCategories();
+  response.render(`edit-post`, {article, categories});
+});
+
+
+articlesRouter.get(URL.ARTICLESURL.ID, async (request, response) => {
+  const article = await api.getArticle(request.params.id);
+  response.render(`post`, {article});
+});
+
+articlesRouter.post(URL.ARTICLESURL.ADD, upload.single(`upload`), async (request, response) => {
+  const {body, file} = request;
+  const articleData = {
+    image: file ? file.filename : null,
+    createDate: body.date,
+    title: body.title,
+    announce: body.announcement,
+    fullText: body[`full-text`],
+    category: await collectCategory(body)
+  };
 
-articlesRouter.get(URL.ARTICLESURL.ID, (request, response) => response.render(`post`));
+  try {
+    await api.createArticle(articleData);
+    response.redirect(URL.MY);
+  } catch (err) {
+    response.redirect(`back`);
+  }
```

Comment:
сейчас дублируется часть логики (вызов `response.redirect`, и если потом редирект будет делать другая функция, нужно будет переименовывать логику в нескольких местах.
гораздо лучше сделать вот так
```suggestion
  let redirectURI = URL.MY;
  try {
    await api.createArticle(articleData);
  } catch (err) {
    redirectURI = `back`;
  }
   response.redirect(redirectURI);
```